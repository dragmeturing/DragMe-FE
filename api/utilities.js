export const BEUrl = 'https://dragme-be.herokuapp.com/api';