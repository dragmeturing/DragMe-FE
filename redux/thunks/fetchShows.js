import { setError, isLoading, getShows } from "../actions/index";

export const fetchShows = () => {
  return async dispatch => {   
    try {
      dispatch(isLoading(true));
      const response = await fetch(`${BEurl}/shows`);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      const shows = await response.json();
      dispatch(getShows(shows.objects));
      dispatch(isLoading(false));
    } catch (error) {
      dispatch(setError(error.error));
    }
  };
};