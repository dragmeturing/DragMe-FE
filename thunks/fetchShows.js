import { setError, isLoading, getShows } from "../actions/index";

export const fetchShows = () => {
  return async dispatch => {
    const url = "";
    //need to add in URL
    try {
      dispatch(isLoading(true));
      const response = await fetch(url);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      const shows = await response.json();
      dispatch(getShows(shows));
      dispatch(isLoading(false));
    } catch (error) {
      dispatch(setError(error.error));
    }
  };
};
