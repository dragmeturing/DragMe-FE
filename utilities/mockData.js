export const shows = [
  {
    id: 1,
    name: "Drag Roulette Show",
    description:
      "Your Friday LATE NIGHT at Mile High Hamburger Mary’s has been upgraded to hysterical! Every 1st and 3rd Friday!",
    event_url:
      "https://www.hamburgermarys.com/denver/events/event/drag-roulette-show/",
    date: "2019-07-14 15:16:38.643787",
    poster_url:
      "https://www.hamburgermarys.com/denver/wp/wp-content/uploads/2019/02/drag-roulette-1024x745.jpg",
    venue_name: "Hamburger Mary's Bar & Grille",
    venue_google_id: "1234556"
  },
  {
    id: 2,
    name: "Drag Roulette Show",
    description:
      "Your Friday LATE NIGHT at Mile High Hamburger Mary’s has been upgraded to hysterical! Every 1st and 3rd Friday!",
    event_url:
      "https://www.hamburgermarys.com/denver/events/event/drag-roulette-show/",
    date: "2019-07-14 15:16:38.643787",
    poster_url:
      "https://www.hamburgermarys.com/denver/wp/wp-content/uploads/2019/02/drag-roulette-1024x745.jpg",
    venue_name: "Hamburger Mary's Bar & Grille",
    venue_google_id: "1234556"
  },
  {
    id: 3,
    name: "Drag Roulette Show",
    description:
      "Your Friday LATE NIGHT at Mile High Hamburger Mary’s has been upgraded to hysterical! Every 1st and 3rd Friday!",
    event_url:
      "https://www.hamburgermarys.com/denver/events/event/drag-roulette-show/",
    date: "2019-07-14 15:16:38.643787",
    poster_url:
      "https://www.hamburgermarys.com/denver/wp/wp-content/uploads/2019/02/drag-roulette-1024x745.jpg",
    venue_name: "Hamburger Mary's Bar & Grille",
    venue_google_id: "1234556"
  }
];

export const venues = [
  {
    id: 1,
    venue_name: "The Clocktower Cabaret",
    venue_google_id: "ChIJASbMldp4bIcRiFeLsW3CUf4"
  },
  {
    id: 2,
    venue_name: "Mile High Hamburger Mary's",
    venue_google_id: "ChIJL4ZMpzN5bIcRe_ttaF-7N3E"
  },
  {
    id: 3,
    venue_name: "Tracks",
    venue_google_id: "ChIJU0l3jxp5bIcRcOzzQKHMBcM"
  },
  {
    id: 4,
    venue_name: "Charlie's Nightclub Denver",
    venue_google_id: "ChIJydsjLC15bIcRcnnY3EIbTgs"
  },
  {
    id: 5,
    venue_name: "Blush & Blu",
    venue_google_id: "ChIJ445gusx-bIcRDpB_himzcuY"
  },
  {
    id: 6,
    venue_name: "X BAR",
    venue_google_id: "ChIJw4sroyx5bIcR5jU8EDoneGU"
  },
  {
    id: 7,
    venue_name: "Pride and Swagger",
    venue_google_id: "ChIJCR82wSt5bIcRsu6dJbKpoX8"
  },
  {
    id: 8,
    venue_name: "Gladys",
    venue_google_id: "ChIJR9gq2Tp_bIcRrtd8dA47as8"
  },
  {
    id: 9,
    venue_name: "#VYBE",
    venue_google_id: "ChIJs4T9jil_bIcRCcy9CMKwJAA"
  }
];
