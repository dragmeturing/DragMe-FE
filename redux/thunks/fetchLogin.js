import { setError, isLoading, loginUser } from "../actions/index";
import { postPerformer } from "../../api/postPerformer";

export const fetchLogin = (token, navigate) => {
  return async dispatch => {
    try {
      dispatch(isLoading(true));
      const response = await postPerformer(token);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      const user = await response.json();
      navigate('User')
      dispatch(loginUser(user));
      dispatch(isLoading(false));
    } catch (error) {
      navigate("User");
      dispatch(setError(error.message));
    }
  };
};
