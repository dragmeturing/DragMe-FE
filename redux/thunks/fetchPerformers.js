import { setError, isLoading, getPerformers } from "../actions/index";
import { BEurl } from "../../utilities/url";

export const fetchPerformers = () => {
  return async dispatch => {
    try {
      dispatch(isLoading(true));
      const response = await fetch(`${BEurl}/performers`);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      const performers = await response.json();
      dispatch(getPerformers(performers));
      dispatch(isLoading(false));
    } catch (error) {
      dispatch(setError(error.message));
    }
  };
};
