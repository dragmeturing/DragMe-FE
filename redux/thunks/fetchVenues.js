import { setError, isLoading, getVenues } from "../actions/index";
import { BEurl } from "../../utilities/url";

export const fetchVenues = () => {
  return async dispatch => {   
    try {
      dispatch(isLoading(true));
      const response = await fetch(`${BEurl}/venues`);
      if (!response.ok) {
        throw Error(response.statusText);
      }
      const venues = await response.json();
      dispatch(getVenues(venues));
      dispatch(isLoading(false));
    } catch (error) {
      dispatch(setError(error.message));
    }
  };
};