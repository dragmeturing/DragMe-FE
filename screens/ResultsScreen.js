import React, { Component } from "react";
import { ScrollView, StyleSheet, ActivityIndicator } from "react-native";
import ResultCard from "../components/ResultCard";
import { mainStyles } from "../constants/mainStyles";
import { primaryColor, accentColor } from "../constants/Colors";
import { fetchShows } from '../thunks/fetchShows';
import { connect } from 'react-redux';

export class ResultsScreen extends Component {
  constructor(props) {
    super(props)
  }

  componentDidMount() {
    this.props.fetchShows()
  };

  render() {
    const { shows } = this.props
    const { container } = mainStyles;
    const results = shows.length 
    ? shows.objects.map(show => (
      <ResultCard
        key={show.id}
        data={show}
        navtigate={this.props.navigation.navigate}
        />
      ))
    : <ActivityIndicator size="large" color={accentColor} />;

    return <ScrollView contentContainerStyle={container}>{results}</ScrollView>;
  };
};

ResultsScreen.navigationOptions = {
  title: "Shows",
  headerStyle: {
    backgroundColor: primaryColor
  }
};


export const mapStateToProps = (state) => {
  console.log('state', state)
  return ({shows: state.shows})
};

export const mapDispatchToProps = (dispatch) => ({
  fetchShows: shows => dispatch(fetchShows(shows))
});

export default connect(mapStateToProps, mapDispatchToProps)(ResultsScreen);