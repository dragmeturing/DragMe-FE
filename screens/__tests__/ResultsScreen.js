import React from 'react';
import { ResultsScreen, mapStateToProps, mapDispatchToProps } from '../ResultsScreen';
import { shallow } from 'enzyme';
import { fetchShows } from '../../redux/thunks/fetchShows';
import { cleanVenues } from '../../utilities/mockData';

jest.mock('../../redux/thunks/fetchShows');

describe('ResultsScreen',() => {
  let wrapper, mockFetchShows, mockFetchVenues, mockShows;

  beforeEach(() => {
    mockFetchShows = jest.fn();
    mockFetchVenues = jest.fn();
    mockShows = [{title: 'new drag', id: 1 },{title: 'drag this', id: 2 }]
    wrapper = shallow(<ResultsScreen
      shows = {mockShows}
      venues={cleanVenues}
      fetchShows={mockFetchShows}
      fetchVenues={mockFetchVenues}
    />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when there are no shows', () => {
     wrapper = shallow(
       <ResultsScreen
         shows={[]}
         fetchShows={mockFetchShows}
         fetchVenues={mockFetchVenues}
       />
     );
    expect(wrapper).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('should call fetchShows', () => {
      expect(mockFetchShows).toHaveBeenCalled();
    });
  });

  describe('mapStateToProps', () => {
    it('should map state to props', () => {
      const mockState = {shows:mockShows};
      const expected = {shows:mockShows};

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected)
    });
  });

    describe('mapDispatchToProps', () => {
    it('should call dispatch with the currect thunk', () => {
      const mockDispatch = jest.fn();

      const props = mapDispatchToProps(mockDispatch);

      const thunk = fetchShows(mockShows);

      props.fetchShows(mockShows);

      expect(mockDispatch).toHaveBeenCalledWith(thunk)
    });
  });
});


