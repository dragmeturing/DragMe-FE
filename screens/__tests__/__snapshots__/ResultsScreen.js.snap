// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsScreen should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`ResultsScreen should match the snapshot when there are no shows 1`] = `ShallowWrapper {}`;
